src/components/component/secrets/azure-key-vault-secret-state-table-row.tsx

```diff
@@ -1,31 +1,86 @@
 import {
   AzureKeyVaultSecretStatusModel,
   AzureKeyVaultSecretStatusModelValidationMap,
 } from '../../../models/azure-key-vault-secret-status';
+import { Duration } from '../../time/duration';
 
 export interface AzureKeyVaultSecretStateTableRowProps {
-  secretInReplica: AzureKeyVaultSecretStatusModel;
+  secretInConsumer: AzureKeyVaultSecretStatusModel;
 }
 
 export const AzureKeyVaultSecretStateTableRow = (
   props: AzureKeyVaultSecretStateTableRowProps
 ): JSX.Element => {
+  const getSecretConsumer = (
+    secretInConsumer: AzureKeyVaultSecretStatusModel
+  ): JSX.Element => {
+    if (
+      secretInConsumer.batchName != null &&
+      secretInConsumer.batchName.length > 0
+    )
+      // show only first secret-version entry for pods of this batch
+      return (
+        <Typography as="span">
+          batch: {smallScheduledBatchName(secretInConsumer.batchName)}
+        </Typography>
+      );
+    if (secretInConsumer.jobName != null && secretInConsumer.jobName.length > 0)
+      return (
+        <Typography as="span">
+          job: {smallScheduledBatchName(secretInConsumer.jobName)}
+        </Typography>
+      );
+    if (secretInConsumer.replicaName.toLowerCase() === 'new jobs')
+      return <Typography as="span">New job</Typography>;
+    return (
+      <Typography as="span">
+        replica: {smallScheduledBatchName(secretInConsumer.replicaName)}
+      </Typography>
+    );
+  };
+  const getSecretConsumerCreated = (
+    secretInConsumer: AzureKeyVaultSecretStatusModel
+  ): JSX.Element => {
+    if (
+      secretInConsumer.batchName != null &&
+      secretInConsumer.batchName.length > 0
+    )
+      return (
+        <Duration start={secretInConsumer.batchCreated} end={new Date()} />
+      );
+    if (
+      secretInConsumer.jobName != null &&
+      secretInConsumer.jobName.length > 0
+    ) {
+      return <Duration start={secretInConsumer.jobCreated} end={new Date()} />;
+    }
+    if (secretInConsumer.replicaName.toLowerCase() === 'new jobs') return <></>;
+    return (
+      <Duration start={secretInConsumer.replicaCreated} end={new Date()} />
+    );
+  };
+
   return (
     <Table.Row>
-      <Table.Cell>{props.secretInReplica.version}</Table.Cell>
+      <Table.Cell>{props.secretInConsumer.version}</Table.Cell>
       <Table.Cell>
         <Typography as="span">
-          {smallReplicaName(props.secretInReplica.replicaName)}
+          {getSecretConsumer(props.secretInConsumer)}
         </Typography>
       </Table.Cell>
+      <Table.Cell>
+        {getSecretConsumerCreated(props.secretInConsumer)}
+      </Table.Cell>
     </Table.Row>
   );
 };
 
 AzureKeyVaultSecretStateTableRow.propTypes = {
-  secretInReplica: PropTypes.shape(AzureKeyVaultSecretStatusModelValidationMap),
+  secretInConsumer: PropTypes.shape(
+    AzureKeyVaultSecretStatusModelValidationMap
+  ),
 } as PropTypes.ValidationMap<AzureKeyVaultSecretStateTableRowProps>;
```

Comment:
Bring stateless functions out of render component.
Some methods can be refactored into proper JSX elements.

Suggested edit of component:
```tsx
import { Table, Typography } from '@equinor/eds-core-react';
import * as PropTypes from 'prop-types';

import { Duration } from '../../time/duration';
import {
  AzureKeyVaultSecretStatusModel,
  AzureKeyVaultSecretStatusModelValidationMap,
} from '../../../models/azure-key-vault-secret-status';
import { smallScheduledBatchName } from '../../../utils/string';

export interface AzureKeyVaultSecretStateTableRowProps {
  secret: AzureKeyVaultSecretStatusModel;
}

const ConsumerSecretName = ({
  secret,
}: {
  secret: AzureKeyVaultSecretStatusModel;
}): JSX.Element => {
  let name: string;
  if (secret.batchName?.length > 0) {
    // show only first secret-version entry for pods of this batch
    name = `batch: ${smallScheduledBatchName(secret.batchName)}`;
  } else if (secret.jobName?.length > 0) {
    name = `job: ${smallScheduledBatchName(secret.jobName)}`;
  } else if (secret.replicaName.toLowerCase() === 'new jobs') {
    name = 'New job';
  } else {
    name = `replica: ${smallScheduledBatchName(secret.replicaName)}`;
  }

  return <Typography as="span">{name}</Typography>;
};

const ConsumerSecretCreated = ({
  secret,
}: {
  secret: AzureKeyVaultSecretStatusModel;
}): JSX.Element => {
  if (secret.batchName?.length > 0) {
    return <Duration start={secret.batchCreated} end={new Date()} />;
  } else if (secret.jobName?.length > 0) {
    return <Duration start={secret.jobCreated} end={new Date()} />;
  } else if (secret.replicaName.toLowerCase() === 'new jobs') {
    return <></>;
  }

  return <Duration start={secret.replicaCreated} end={new Date()} />;
};

export const AzureKeyVaultSecretStateTableRow = ({
  secret,
}: AzureKeyVaultSecretStateTableRowProps): JSX.Element => (
  <Table.Row>
    <Table.Cell>{secret.version}</Table.Cell>
    <Table.Cell>
      <Typography as="span">
        <ConsumerSecretName secret={secret} />
      </Typography>
    </Table.Cell>
    <Table.Cell>
      <ConsumerSecretCreated secret={secret} />
    </Table.Cell>
  </Table.Row>
);

AzureKeyVaultSecretStateTableRow.propTypes = {
  secret: PropTypes.shape(AzureKeyVaultSecretStatusModelValidationMap),
} as PropTypes.ValidationMap<AzureKeyVaultSecretStateTableRowProps>;
```

Reply:
Applied